cpp/tensorrt_llm/common/stringUtils.cpp

```diff
@@ -26,14 +26,16 @@
 namespace tensorrt_llm::common
 {
 
-void fmtstr_(char const* format, fmtstr_allocator alloc, void* target, va_list args)
+void fmtstr_(char const* format, fmtstr_allocator alloc, void* target, va_list_guard& args_guard)
 {
     va_list args0;
-    va_copy(args0, args);
+    va_copy(args0, args_guard.get());
+    va_list_guard args0_guard(args0);
```

Comment:
We don't need guard for `args0`. `args0` is no longer used after the first `std::vsnprintf`, and `std::vsnprintf` does not throw.